robot-command/src/main/java/org/obolibrary/robot/FilterCommand.java

```diff
@@ -128,49 +128,81 @@ public CommandState execute(CommandState state, String[] args) throws Exception
     }
     boolean trim = CommandLineHelper.getBooleanValue(line, "trim", true);
 
+    // Get the output IRI for the new ontology
+    IRI outputIRI;
+    String outputIRIString = CommandLineHelper.getOptionalValue(line, "ontology-iri");
+    if (outputIRIString != null) {
+      outputIRI = IRI.create(outputIRIString);
+    } else {
+      // If it is not provided, copy the input IRI
+      outputIRI = inputOntology.getOntologyID().getOntologyIRI().orNull();
+    }
+
+    // Create the output ontology
+    OWLOntology outputOntology;
+    if (outputIRI != null) {
+      outputOntology = manager.createOntology(outputIRI);
+    } else {
+      outputOntology = manager.createOntology();
+    }
+
     // Selects should be processed in order, allowing unions in one --select
     List<List<String>> selectGroups = new ArrayList<>();
     boolean includeAnnotations = false;
+    boolean hadSelection = false;
+
     for (String select : selects) {
       // The single group is a split of the one --select
       List<String> selectGroup = CommandLineHelper.splitSelects(select);
+      // Annotations is a special keyword that applies to all selected objects
       if (selectGroup.contains("annotations")) {
         includeAnnotations = true;
         selectGroup.remove("annotations");
       }
-      selectGroups.add(selectGroup);
-    }
-
-    // If there are no objects, add all the objects from the ontology
-    if (objects.isEmpty()) {
-      objects.addAll(OntologyHelper.getObjects(ontology));
+      // The ontology keyword retrieves the ontology annotations
+      if (selectGroup.contains("ontology")) {
```

Comment:
What about "imports"?

Reply:
The filter command doesn't work on imports (for remove, it just removes the import statements). I guess we could add it in filter to just filter for the import statements...

Reply:
I added `--select imports` for filter - it will copy the import declarations from the input to the output ontology.